templates/cicd/pipeline_cfn.yml

```diff
@@ -140,6 +153,22 @@ Resources:
         Statement:
           - Effect: Allow
             Action:
+              - codestar-connections:CreateConnection
+              - codestar-connections:DeleteConnection
+              - codestar-connections:GetConnection
+              - codestar-connections:ListConnections
+              - codestar-connections:GetIndividualAccessToken
+              - codestar-connections:GetInstallationUrl
+              - codestar-connections:ListInstallationTargets
+              - codestar-connections:StartOAuthHandshake
+              - codestar-connections:UpdateConnectionInstallation
+              - codestar-connections:UseConnection
+              - codestar-connections:RegisterAppCode
+              - codestar-connections:StartAppRegistrationHandshake
+              - codestar-connections:StartUploadArchiveToS3
+              - codestar-connections:GetUploadArchiveToS3Status
+              - codestar-connections:PassConnection
+              - codestar-connections:PassedToService
```

Comment:
Can we split up the codestar permissions into a separate statement if the provider requires it? Something like:
```yaml
{{if .Source.RequiresCodeStarConnections}}
- Effect: Allow
  Action: 
  - ...
  Resource:
  - !Ref SourceConnection
{{- end}}
```
to limit it to our connection

Reply:
Good idea. Bc CodeCommit is (soon-to-be) our only non-CodeStar source, I used
`          {{- if ne .Source.ProviderName "CodeCommit"}}`
I built a CC and a BB pipeline and checked the CFN templates. The resources were applied as anticipated.